tests/integration/trace/test_zipkin_trace.py

```diff
@@ -176,3 +178,17 @@ def hello(request):
 
     yield f
     assert count[0] == 1
+
+
+@pytest.mark.gen_test
+def test_zipkin_trace_sampling(trace_server):
+    run_times = 100000
+    sample_rate = 1.0 - random.random()
+    hook = ZipkinTraceHook(sample_rate=sample_rate)
+    count = 0
+    for _ in range(run_times):
+        if hook._lucky(_uniq_id()):
+            count += 1
+
+    assert 0.9 * run_times * sample_rate <= count
```

Comment:
This test in inadequate. Replace this with:

- A test that ensures that no traces are emitted if rate is 0 even if trace is set to True
- A test that mocks `_lucky` to emit True for a specific case and checks that the trace is emitted. (We want to avoid mocking internal APIs usually but since in this case we can't control randomness cleanly, we have to do this.)